compiler/aarch64/codegen/ARM64Instruction.hpp

```diff
@@ -106,6 +106,8 @@ namespace TR
 class ARM64ImmInstruction : public TR::Instruction
    {
    uint32_t _sourceImmediate;
+   TR_ExternalRelocationTargetKind _reloKind;
+   TR::SymbolReference *_symbolReference;
```

Comment:
How is an `ARM64ImmInstruction` with a SymbolReference different than an `ARM64ImmSymInstruction`?  As a developer, how would I know which form to pick?

Reply:
The main difference from `ARM64ImmSymInstruction` is `TR_ExternalRelocationTargetKind`.
Power uses the `ImmInstruction` with `TR_ExternalRelocationTargetKind` in the following lines.  The first one is without `SymbolReference`, and the other is with `SymbolReference`:
https://github.com/eclipse/omr/blob/f2285166b44222b41ecb7a73422180518a290028/compiler/p/codegen/OMRCodeGenerator.cpp#L409-L410
https://github.com/eclipse/omr/blob/f2285166b44222b41ecb7a73422180518a290028/compiler/p/codegen/OMRCodeGenerator.cpp#L425-L427

Reply:
How about making a new class `ARM64RelocatableImmInstruction`, to make the difference clearer from `ARM64ImmInstruction` and `ARM64ImmSymInstruction`?